ml-agents/mlagents/trainers/learn.py

```diff
@@ -198,26 +192,21 @@ def write_timing_tree(output_dir: str) -> None:
         )
 
 
-def create_sampler_manager(sampler_config, run_seed=None):
-    resample_interval = None
+def maybe_add_samplers(
+    sampler_config: Optional[Dict], env: SubprocessEnvManager, run_seed: int
+) -> None:
+    """
+    Adds samplers to env if sampler config provided and sets seed if not configured.
+    :param sampler_config: validated dict of sampler configs. None if not included.
+    :param env: env manager to pass samplers via reset
+    :param run_seed: Random seed used for training.
+    """
     if sampler_config is not None:
-        if "resampling-interval" in sampler_config:
-            # Filter arguments that do not exist in the environment
-            resample_interval = sampler_config.pop("resampling-interval")
-            if (resample_interval <= 0) or (not isinstance(resample_interval, int)):
-                raise SamplerException(
-                    "Specified resampling-interval is not valid. Please provide"
-                    " a positive integer value for resampling-interval"
-                )
-
-        else:
-            raise SamplerException(
-                "Resampling interval was not specified in the sampler file."
-                " Please specify it with the 'resampling-interval' key in the sampler config file."
-            )
-
-    sampler_manager = SamplerManager(sampler_config, run_seed)
-    return sampler_manager, resample_interval
+        # If the seed is not specified in yaml, this will grab the run seed
+        for _, v in sampler_config.items():
+            if v.seed == -1:
+                v.seed = run_seed
```

Comment:
I think this could be bad; it could mean multiple sampled variables with the same seed, so the results would be highly correlated.

How about something like
```
for offset, v in enumerate(sampler_config.values()):
  ...
  v.seed = run_seed + offset
```

Reply:
good catch